app/src/main/java/com/rob729/quiethours/Fragments/NewProfileFragment.kt

```diff
@@ -142,7 +142,11 @@ class NewProfileFragment : Fragment() {
                 while (i < 7) {
                     if (days[i]) {
                         sAlarm(i + 1, profile)
-                        eAlarm(i + 1, profile)
+                        if ((shr > ehr) || ((shr == ehr) && (smin > emin))) {
+                            eAlarm(i + 2, profile)
```

Comment:
when i=6 and i+2=8, then do you think it will work?
Have a look at it.

Reply:
I am so sorry, that condition completely slipped off my mind. I'll just put in another if-else, cause i can't change value of 'i' and other option is to declare another variable which will just increase app's size.